src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1089,6 +1061,63 @@ private TokenValidationResult ValidateToken(string token, JsonWebToken outerToke
+            }
+            catch (Exception ex)
+            {
+                IOException ioException;
+                if ((ioException = IsIOExceptionWithFailureInformation(ex)) != null)
+                {
+                    // Retry configuration retrieval on network failure (ServiceUnavailable or RequestTimeout).
+                    if (ioException.Data[HttpResponseConstants.StatusCode].Equals(HttpStatusCode.RequestTimeout) || ioException.Data[HttpResponseConstants.StatusCode].Equals(HttpStatusCode.ServiceUnavailable))
+                    {
+                        LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10265, ioException.Data[HttpResponseConstants.StatusCode], ioException.Data[HttpResponseConstants.ResponseContent], validationParameters.ConfigurationManager.MetadataAddress));
+                        try
+                        {
+                            currentConfiguration = validationParameters.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
+                            return currentConfiguration;
+                        }
+                        catch (Exception retryException)
+                        {
+                            if ((ioException = IsIOExceptionWithFailureInformation(retryException)) != null)
+                                LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10266, validationParameters.ConfigurationManager.MetadataAddress, ioException.Data[HttpResponseConstants.StatusCode], ioException.Data[HttpResponseConstants.ResponseContent], retryException.ToString()));
+                            else
+                                LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10261, validationParameters.ConfigurationManager.MetadataAddress, ex.ToString()));
+                        }
+                    }
+                    else
+                    {
+                        LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10266, validationParameters.ConfigurationManager.MetadataAddress, ioException.Data[HttpResponseConstants.StatusCode], ioException.Data[HttpResponseConstants.ResponseContent], ex.ToString()));
+                    }
+                }
+                else
+                {
+                    // The exception is not re-thrown here as the TokenValidationParameters may have the issuer and signing key set
+                    // directly on them.
+                    LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10261, validationParameters.ConfigurationManager.MetadataAddress, ex.ToString()));
+                }
+            }
+
+            return null;
+        }
+
+        private static IOException IsIOExceptionWithFailureInformation(Exception ex)
+        {
+            if (ex.GetType().Equals(typeof(InvalidOperationException)) && ex.Message.Contains("IDX20803 :")
```

Comment:
should we have a space after IDX20803 before the : ?

Reply:
I've removed the check for IDX20803 based on Danny's comments in https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1784. 

For the sake of easier review, I've merged this PR into the PR above since it touches the same code and closed this one. You will find these comments addressed there.